### DISCUSSION TITLE
Can't boot my Alienware M18

### DISCUSSION THREAD

**Discussion by philmb3487** [zbm-dev/zfsbootmenu#689] [Questions] [answered]
Short story, my laptop exploded and I had to get a new one. My Alienware M18 R2 is not booting ZBM, the screen just doesn't change and then nothing happens.
I can confirm the Linux EFI stub works,  I did something like cp /boot/vmlinuxblah /mnt/EFI/BOOT/BOOTX64.EFI and that fired up the kernel.

What can I do now to troubleshoot this issue?

**Comment by philmb3487** [comment]
Sorry If there's still typos all over the place, I'm not use to this keyboard yet.

**Comment by ahesford** [comment]
The release images use the old gummiboot EFI stub, which is known to have problems with some hardware. I suggest you:
1. Install rEFInd or systemd-boot
2. Download the [ZBM component archive](https://get.zfsbootmenu.org/latest.tar.gz)
3. Unpack the archive to your EFI system partition at, *e.g.*, `EFI/zfsbootmenu`
4. Configure rEFInd or systemd-boot to launch the components

**Comment by philmb3487** [comment]
I would rather not mess with rEFInd or systemd (I don't use either and they're big complex systems, especially systemd)

Could you build me an image with kernel 6.11? I tried building,

`./releng/docker/image-build.sh -k 6.11 master` but there's nothing in my build directory and I don't see the image in my docker store, and I'm too dumb to understand the documentation.

Thanks

**Reply by classabbyamp** [reply]
systemd-boot is just gummiboot under a systemd-prefixed name, it doesn't need all of systemd to work

**Comment by philmb3487** [comment]
Actually why are there 6 different build scripts?

there's image-build, build-init, zbm-builder, generate-zbm, and it's not explained anywhere how they differ.

**Reply by zdykstra** [reply]
https://docs.zfsbootmenu.org/en/v2.3.x/general/container-building.html

**Reply by philmb3487** [reply]
Oh the buildah thing has its own repository parallel to Docker (why?), I see it there, the image is 0fcd95aa1cbf.
So then I plug this into the main builder, `./zbm-builder.sh -i 0fcd95aa1cbf`

But it's still not seeing the image. I don't know.

**Reply by classabbyamp** [reply]
you mean ghcr.io? that's just github's OCI container repository, no different than dockerhub except they [don't threaten to end free image hosting](https://www.docker.com/blog/we-apologize-we-did-a-terrible-job-announcing-the-end-of-docker-free-teams/)

**Reply by ahesford** [reply]
There aren't six different build scripts.
- You should generally stay ouf of `releng/docker`. The `image-build.sh` is a buildah script to prepare a container image, and `build-init.sh` is the container entrypoint.
- `generate-zbm` is the utility that actually prepares a ZFSBootMenu image. You use this to build a custom image on the host.
- `zbm-builder.sh` is a convenience wrapper for invoking the build container, which internally runs `generate-zbm` in a controlled environment.

**Comment by philmb3487** [comment]
Well, but i need to run the releng first, because I can’t find any arguments for setting kernel versions in the main builder.

What’s a convenience wrapper?
How do I pass the build image to generate zbm?

**Comment by ahesford** [comment, marked as answer]
You almost certainly do not need a newer kernel version in ZFSBootMenu. If you run

```sh
mkdir -p zbm-builds
/path/to/zbm-builder.sh -i ghcr.io/zbm-dev/zbm-builder:development -b zbm-builds
```

you should find new default images in `zbm-builds/build`. The EFI executable in that directory will have been built with the newer `systemd-boot` stub.

**Comment by philmb3487** [comment]
That’s it! That worked with the newer 6.6 series.
Do you know how to generate locally without relying on the docker hub image?
I would like to do that, if you know how to do it?

Thanks a million!
